src/app/services/actions/adaptive-cards-action-creator.ts

```diff
@@ -0,0 +1,82 @@
+import { IAction } from '../../../types/action';
+import { IQuery } from '../../../types/query-runner';
+import { ADAPTIVE_FETCH_ERROR , ADAPTIVE_FETCH_PENDING, ADAPTIVE_FETCH_SUCCESS } from '../redux-constants';
+
+export function getAdaptiveCardSuccess(result: string): IAction {
+  return {
+    type: ADAPTIVE_FETCH_SUCCESS,
+    response: result,
+  };
+}
+
+export function getAdaptiveCardError(error: string): IAction {
+  return {
+    type: ADAPTIVE_FETCH_ERROR,
+    response: error
+  };
+}
+
+export function getAdaptiveCardPending(): IAction {
+  return {
+    type: ADAPTIVE_FETCH_PENDING,
+    response: ''
+  };
+}
+
+export function getAdaptiveCard(payload: string, sampleQuery: IQuery): Function {
+  return async (dispatch: Function) => {
+
+    if (!payload) {
+      // no payload so return empty result
+      return dispatch(getAdaptiveCardSuccess(''));
+    }
+
+    const template = lookupTemplate(sampleQuery);
+    if (template === '' ) {
+      // we dont support this card yet
+      return dispatch(getAdaptiveCardError('No template available'));
+    }
+
+    const url = `https://templates.adaptivecards.io/graph.microsoft.com/${template}`;
+    const body = JSON.stringify(payload);
+    dispatch(getAdaptiveCardPending());
+
+    return fetch(url, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body
+    }).then(resp => resp.json())
+      .then((result) => {
+        if (result.error) {
+          throw (result.error);
+        }
+        // give back the result of the card
+        return dispatch(getAdaptiveCardSuccess(result));
+      })
+      .catch(error => {
+        // something wrong happened
+        return dispatch(getAdaptiveCardError(error));
+      });
+  };
+}
+
+function lookupTemplate(sampleQuery: IQuery) {
+  if (sampleQuery) {
+    const urlObject: URL = new URL(sampleQuery.sampleUrl);
+    // remove the prefix i.e. beta or v1.0 and any possible extra whack character at the end'/'
+    const query = urlObject.pathname.substr(6).replace(/\/$/, '') + urlObject.search;
+    const template = templateTable[query];
+    if (template) {
+      return template;
+    }
+  }
```

Comment:
Do you think it would be better to use the return first approach here?
```js
if (!sampleQuery) { return }

...
```
Also you can specify the return type.

Reply:
+1 on specifying the return type. 

Returning first is rendered moot with `return templateTable[query]` raised in later comment